roles/accprovision/tasks/process.yaml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Create temporary directory for acc-provision run
+  file:
+    path: "{{ acc_provision_dir_path }}"
+    state: directory
+    mode: '0777'
+
+- name: Get acc-provision configmap
+  k8s_info:
+    api_version: v1
+    kind: ConfigMap
+    name: acc-provision-config
+    namespace: "{{ lookup('env', 'WATCH_NAMESPACE') }}"
+  register: cmap_data
+
+- name: Set fact
+  set_fact:
+    cmap_content: "{{ cmap_data['resources'][0]}}"
+
+- name: Write cmap YAML
+  copy:
+    dest: "{{ acc_provision_dir_path }}/cmap.yaml"
+    content: "{{ cmap_content }}"
+
+- name: Set input YAML
+  set_fact:
+    cmap_input: "{{ cmap_data['resources'][0]['data']['spec'] }}"
+
+- name: Get CRD definition for accprovisioninput
+  community.kubernetes.k8s_info:
+    api_version: v1
+    kind: CustomResourceDefinition
+    name: accprovisioninputs.aci.ctrl
+  register: raw_crd
+
+- name: print
+  set_fact:
+    crd_definition: "{{ raw_crd['resources'][0]['spec']['versions'][0]['schema']['openAPIV3Schema']['properties']['spec']['properties']['acc_provision_input']['properties'] }}"
+
+- name: Compute acc-provision input file
+  script: "scripts/process_string_to_yaml.py"
+  args:
+    executable: /usr/bin/python3
+  environment:
+    CRDINPUT: "{{ acc_provision_input }}"
+    CRD_DEFINITION: "{{ crd_definition }}"
+    CONFIGMAP_INPUT_FILE: "{{ acc_provision_dir_path }}/cmap.yaml"
+    ACCPROVDIR: "{{ acc_provision_dir_path }}"
+    ACCPROVFILE: "{{ acc_provision_file_name }}"
+  register: hello
+
+- name: Delete the configmap
+  k8s:
+    state: absent
+    api_version: v1
+    kind: ConfigMap
+    name: acc-provision-config
+    namespace: "{{ lookup('env', 'WATCH_NAMESPACE') }}"
+    wait: yes
+
+- name: Create new configmap
+  k8s:
+    state: present
+    force: yes
+    src: "{{ acc_provision_dir_path }}/cmap.yaml"
+
+- name: Generate acicnioperator CR
```

Comment:
Does this check if acicnioperator CR is already created or not. Will it overwrite?

Reply:
We have now switched to creating a deployment file- and the k8s module in Ansible patches the YAML objects. https://docs.ansible.com/ansible/latest/collections/community/kubernetes/k8s_module.html#parameter-state